### PR TITLE
remove old code, fix sdk version, update action

### PR DESCRIPTION
**Pull request by m1ga**
* update github action
* remove <= Ti 10 if cases
* remove `ti config` selected.sdk check (not used anymore)
* add ti sdk version to the test tiapp.xml file
* use 12.5.0 in the github action so it will work with the test

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,12 +11,12 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v4
         with:
-          node-version: '16.x'
+          node-version: '18.x'
```
Comment (cb1kenobi) [inline .github/workflows/build.yml:19]: Please use Node 22. We are dropping Node 18 in the next major CLI release.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -8,14 +8,14 @@ jobs:
...
+      uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
     - name: Setup node
-      uses: actions/setup-node@v2
+      uses: actions/setup-node@v4
       with:
-        node-version: '16'
+        node-version: '18'
```
Comment (cb1kenobi) [inline .github/workflows/publish.yml:18]: Same, let's use Node 22 please.

File: test/projects/HarnessTemplate/tiapp.xml [hunk 3]
```diff
@@ -69,4 +69,5 @@
         <module platform="android">ti.map</module>
         <module platform="iphone">ti.map</module>
     </modules>
+    <sdk-version>12.5.0.GA</sdk-version>
```
Comment (cb1kenobi) [inline test/projects/HarnessTemplate/tiapp.xml:72]: If you omit `<sdk-version>`, it'll use the latest installed SDK. This is handy so we don't have to remember to bump this.

**Review by cb1kenobi — CHANGES_REQUESTED** [review]
I love this PR! Needs a couple of small changes.

What's up with the removal of `Ti.UI.iPhone.NavigationGroup.js`? Is this a breaking change?

File: test/projects/HarnessTemplate/tiapp.xml (hunk 3 above)
Comment (m1ga) [reply test/projects/HarnessTemplate/tiapp.xml:72]: The test failed without a version and that's why I've put 12.5.0 in there and made sure 12.5.0 is installed in the build workflow

Comment (cb1kenobi) [reply test/projects/HarnessTemplate/tiapp.xml:72]: The Titanium CLI doesn't need a `<sdk-version>`. I built an alloy app without a `<sdk-version>` and it also worked.

I tried running the tests on Windows and they blow up saying `'alloy' is not recognized as an internal or external command`. `alloy` is in the PATH. I haven't dug further into it.

Maybe leave this as `latest`?

Comment (m1ga) [reply test/projects/HarnessTemplate/tiapp.xml]: This is the error when I use latest and remove the `<sdk-version>` again:

Bildschirmfoto_20241007_104735

```
Error:  "undefined" was found in the "sdk-version" field of your tiapp.xml.
Error:  If you are building with the legacy titanium.py script and are specifying 
Error:  an SDK version as a CLI argument that is different than the one in your 
Error:  tiapp.xml, please change the version in your tiapp.xml file.
```

Comment (cb1kenobi) [reply test/projects/HarnessTemplate/tiapp.xml]: I'll test on Linux when I get a sec.

Comment (cb1kenobi) [reply test/projects/HarnessTemplate/tiapp.xml]: Ok, looks like you backed out the changes, so we're good.

**Comment by cb1kenobi** [comment]
The tests fail on WSL for the same reason the CI checks fail below. For example:

```
alloy compile widgets/wpath/ [ios] compiles without critical error
  Expected { code : 1, killed : false, signal : null, cmd : 'alloy compile /home/chris/projects/titanium/alloy/test/projects/Harness --config platform=ios' } to be falsy.
```

Not sure what's causing that. Isn't the Android and iOS stuff mocked?

**Review by cb1kenobi — APPROVED** [review]

**Comment by m1ga** [comment]
@cb1kenobi did you had some time to look at this? I can put in the fixed <sdk-version> again so the test will succeed

**Comment by m1ga** [comment]
@cb1kenobi I've added the `sdk-version` back to the test tiapp.xml so everything is green and we can get this merged + #1420 (and #1410 in the future as it is very hand to get the database if you work with models :) ).

**Review by cb1kenobi — APPROVED** [review]

**Pull request merged**